coderadar-test/src/test/java/io/reflectoring/coderadar/rest/useradministration/permissions/SetUserRoleForProjectControllerIntegrationTest.java

```diff
@@ -0,0 +1,109 @@
+package io.reflectoring.coderadar.rest.useradministration.permissions;
+
+import static io.reflectoring.coderadar.rest.JsonHelper.fromJson;
+import static org.springframework.restdocs.payload.PayloadDocumentation.requestFields;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import io.reflectoring.coderadar.graph.projectadministration.domain.ProjectEntity;
+import io.reflectoring.coderadar.graph.projectadministration.project.repository.ProjectRepository;
+import io.reflectoring.coderadar.graph.useradministration.domain.UserEntity;
+import io.reflectoring.coderadar.graph.useradministration.repository.UserRepository;
+import io.reflectoring.coderadar.rest.ControllerTestTemplate;
+import io.reflectoring.coderadar.rest.ProjectRoleJsonWrapper;
+import io.reflectoring.coderadar.rest.domain.ErrorMessageResponse;
+import io.reflectoring.coderadar.useradministration.domain.ProjectRole;
+import io.reflectoring.coderadar.useradministration.service.security.PasswordUtil;
+import java.util.Date;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+class SetUserRoleForProjectControllerIntegrationTest extends ControllerTestTemplate {
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private ProjectRepository projectRepository;
+
+  private ProjectEntity testProject;
+  private UserEntity testUser;
+
+  @BeforeEach
+  void setUp() {
+    testProject = new ProjectEntity();
+    testProject.setVcsUrl("https://valid.url");
+    testProject.setName("project");
+    testProject.setVcsStart(new Date());
+    testProject.setVcsOnline(true);
+    testProject.setVcsPassword("testPassword");
+    testProject.setVcsUsername("testUser");
+    projectRepository.save(testProject);
+
+    testUser = new UserEntity();
+    testUser.setUsername("username");
+    testUser.setPassword(PasswordUtil.hash("password1"));
+    userRepository.save(testUser);
+  }
+
```

Comment:
What happens if there is no user with this id for this project? What happens if user and project do not exist?